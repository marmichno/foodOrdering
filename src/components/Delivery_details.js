import {Link} from 'react-router-dom';

export const Delivery_details = () => {

    return(
        <div className="deliveryDetailsContainer">
            <div className="formContainer">
                <div className="nameContainer">
                    <div>
                        <label for="firstName">
                            <span className="contentName"><p>First Name</p></span>
                        </label>
                        <input type="text" autocomplete="off" name="category" className="firstSentenceInput" required></input>  
                    </div>

                    <div>
                    <label for="secondName">
                        <span className="contentName"><p>Last Name</p></span>
                    </label>
                    <input type="text" autocomplete="off" name="category" className="firstSentenceInput" required></input>
                    </div>
                </div>

                <label for="category">
                    <span className="contentName"><p>Phone number</p></span>
                </label>
                <input type="tel" autocomplete="off" name="category" className="firstSentenceInput" required></input>

                <label for="category">
                    <span className="contentName"><p>City</p></span>
                </label>
                <input type="tel" autocomplete="off" name="category" className="firstSentenceInput" required></input>

                <label for="category">
                    <span className="contentName"><p>Addres</p></span>
                </label>
                <input type="tel" autocomplete="off" name="category" className="firstSentenceInput" required></input>

                <label for="category">
                    <span className="contentName"><p>Delivery time</p></span>
                </label>
                <input type="time" autocomplete="off" name="category" className="firstSentenceInput" required></input>

                <label for="category">
                    <span className="contentName"><p>Additional notes</p></span>
                </label>
                <textarea autocomplete="off" name="category" className="firstSentenceInput" required></textarea>

                <Link to={{
                        pathname:'/order/choose_delivery/delivery_details/delivery_time'
                }}>
                <button>Confirm order details</button>
                </Link>
            </div>
        </div>
    )   
}