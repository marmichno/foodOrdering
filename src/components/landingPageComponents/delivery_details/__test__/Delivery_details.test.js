import {Delivery_details} from '../Delivery_details';
import {BrowserRouter, Router} from 'react-router-dom';
import {render, fireEvent, waitFor} from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import { createMemoryHistory } from "history";
import * as redux from 'react-redux';
import TestRenderer from "react-test-renderer";


const store = createStore(() => [], {}, applyMiddleware());

test("input is visible on deliveryType delviery", () => {

    const spy = jest.spyOn(redux, 'useSelector');
    spy.mockReturnValue([]);

    const history = createMemoryHistory();
    const location = {
        pathname: '/order/choose_delivery/delivery_details',
        state: {
            deliveryType:"delivery"
        }
    }
    history.push(location);


    const {getAllByText} = render(<BrowserRouter><Provider store={store}>
    <Delivery_details location={history.location}/>
    </Provider></BrowserRouter>);


    expect(getAllByText('City'));

});

test("Input isnt visible on delivery type pickUp", () => {

    const spy = jest.spyOn(redux, 'useSelector');
    spy.mockReturnValue([]);

    const history = createMemoryHistory();
    const location = {
        pathname: '/order/choose_delivery/delivery_details',
        state: {
            deliveryType:"pickUp"
        }
    }
    history.push(location);


    const {queryByTestId} = render(<BrowserRouter><Provider store={store}>
    <Delivery_details location={history.location}/>
    </Provider></BrowserRouter>);

    expect(queryByTestId("city")).toBeNull();

});

test("correct subtotal value", () => {
    const spy = jest.spyOn(redux, 'useSelector');
    spy.mockReturnValue([{ 
        price:"6.00",
        name:"philadelphia",
        quantity:5
     }]);

    const history = createMemoryHistory();
    const location = {
        pathname: '/order/choose_delivery/delivery_details',
        state: {
            deliveryType:"pickUp"
        }
    }
    history.push(location);


    const {getByText} = render(<BrowserRouter><Provider store={store}>
    <Delivery_details location={history.location}/>
    </Provider></BrowserRouter>);

    expect(getByText('Total amount: 30.00$'));

});

test("user may only choose one checkbox", async () => {
    const spy = jest.spyOn(redux, 'useSelector');
    spy.mockReturnValue([{ 
        price:"6.00",
        name:"philadelphia",
        quantity:5
     }]);

    const history = createMemoryHistory();
    const location = {
        pathname: '/order/choose_delivery/delivery_details',
        state: {
            deliveryType:"pickUp"
        }
    }
    history.push(location);

    const {getByTestId} = render(<BrowserRouter><Provider store={store}>
    <Delivery_details location={history.location}/>
    </Provider></BrowserRouter>);

    const paymentInAdvance = getByTestId("paymentInAdvance");
    const paymentOnDelivery = getByTestId("paymentOnDelivery");

    fireEvent.click(paymentOnDelivery);
    fireEvent.click(paymentInAdvance);

    await waitFor(() => {
        expect(paymentInAdvance).toHaveClass('active');
        expect(paymentOnDelivery).not.toHaveClass('active');
    });
});


test("matches snapshot", () => {
    const spy = jest.spyOn(redux, 'useSelector');
    spy.mockReturnValue([{ 
        price:"6.00",
        name:"philadelphia",
        quantity:5
     }]);

    const history = createMemoryHistory();
    const location = {
        pathname: '/order/choose_delivery/delivery_details',
        state: {
            deliveryType:"pickUp"
        }
    }
    history.push(location);

    const tree = TestRenderer.create(<BrowserRouter><Provider store={store}><Delivery_details location={history.location}/></Provider></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
});

test("matches snapshot2", () => {
    const spy = jest.spyOn(redux, 'useSelector');
    spy.mockReturnValue([{ 
        price:"6.00",
        name:"philadelphia",
        quantity:5
     }]);

    const history = createMemoryHistory();
    const location = {
        pathname: '/order/choose_delivery/delivery_details',
        state: {
            deliveryType:"delivery"
        }
    }
    history.push(location);

    const tree = TestRenderer.create(<BrowserRouter><Provider store={store}><Delivery_details location={history.location}/></Provider></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
});