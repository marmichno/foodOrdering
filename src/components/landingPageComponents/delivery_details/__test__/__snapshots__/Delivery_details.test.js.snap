// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="deliveryDetailsContainer"
>
  <div
    className="formContainer"
    style={
      Object {
        "transform": "translateX(2000px) translateZ(0)",
      }
    }
  >
    <div
      className="nameContainer"
    >
      <div>
        <label
          htmlFor="firstName"
        >
          <span
            className="contentName"
          >
            <p>
              First Name
            </p>
          </span>
        </label>
        <input
          autoComplete="off"
          className="firstSentenceInput"
          name="category"
          required={true}
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="secondName"
        >
          <span
            className="contentName"
          >
            <p>
              Last Name
            </p>
          </span>
        </label>
        <input
          autoComplete="off"
          className="firstSentenceInput"
          name="category"
          required={true}
          type="text"
        />
      </div>
    </div>
    <label
      htmlFor="category"
    >
      <span
        className="contentName"
      >
        <p>
          Phone number
        </p>
      </span>
    </label>
    <input
      autoComplete="off"
      className="firstSentenceInput"
      name="category"
      required={true}
      type="tel"
    />
    <label
      htmlFor="category"
    >
      <span
        className="contentName"
      >
        <p>
          Additional notes
        </p>
      </span>
    </label>
    <textarea
      autoComplete="off"
      className="firstSentenceInput"
      name="category"
      required={true}
    />
    <div
      className="formCheckboxes"
    >
      <div>
        <p>
          Payment in advance
        </p>
        <svg
          data-paymenttype="advance"
          data-testid="paymentInAdvance"
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1.1"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14 0h-12c-1.1 0-2 0.9-2 2v12c0 1.1 0.9 2 2 2h12c1.1 0 2-0.9 2-2v-12c0-1.1-0.9-2-2-2zM7 12.414l-3.707-3.707 1.414-1.414 2.293 2.293 4.793-4.793 1.414 1.414-6.207 6.207z"
          />
        </svg>
      </div>
      <div>
        <p>
          Payment on delivery
        </p>
        <svg
          data-paymenttype="delivery"
          data-testid="paymentOnDelivery"
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1.1"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14 0h-12c-1.1 0-2 0.9-2 2v12c0 1.1 0.9 2 2 2h12c1.1 0 2-0.9 2-2v-12c0-1.1-0.9-2-2-2zM7 12.414l-3.707-3.707 1.414-1.414 2.293 2.293 4.793-4.793 1.414 1.414-6.207 6.207z"
          />
        </svg>
      </div>
    </div>
    <h1>
      Total amount: 
      0.00
      $
    </h1>
    <a
      href="/order/choose_delivery/delivery_details/delivery_time"
      onClick={[Function]}
    >
      <button>
        Confirm order details
      </button>
    </a>
  </div>
</div>
`;

exports[`matches snapshot2 1`] = `
<div
  className="deliveryDetailsContainer"
>
  <div
    className="formContainer"
    style={
      Object {
        "transform": "translateX(2000px) translateZ(0)",
      }
    }
  >
    <div
      className="nameContainer"
    >
      <div>
        <label
          htmlFor="firstName"
        >
          <span
            className="contentName"
          >
            <p>
              First Name
            </p>
          </span>
        </label>
        <input
          autoComplete="off"
          className="firstSentenceInput"
          name="category"
          required={true}
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="secondName"
        >
          <span
            className="contentName"
          >
            <p>
              Last Name
            </p>
          </span>
        </label>
        <input
          autoComplete="off"
          className="firstSentenceInput"
          name="category"
          required={true}
          type="text"
        />
      </div>
    </div>
    <label
      htmlFor="category"
    >
      <span
        className="contentName"
      >
        <p>
          Phone number
        </p>
      </span>
    </label>
    <input
      autoComplete="off"
      className="firstSentenceInput"
      name="category"
      required={true}
      type="tel"
    />
    <label
      htmlFor="category"
    >
      <span
        className="contentName"
        data-testid="city"
      >
        <p>
          City
        </p>
      </span>
    </label>
    <input
      autoComplete="off"
      className="firstSentenceInput"
      name="category"
      required={true}
      type="tel"
    />
    <label
      htmlFor="category"
    >
      <span
        className="contentName"
      >
        <p>
          Addres
        </p>
      </span>
    </label>
    <input
      autoComplete="off"
      className="firstSentenceInput"
      name="category"
      required={true}
      type="tel"
    />
    <label
      htmlFor="category"
    >
      <span
        className="contentName"
      >
        <p>
          Additional notes
        </p>
      </span>
    </label>
    <textarea
      autoComplete="off"
      className="firstSentenceInput"
      name="category"
      required={true}
    />
    <div
      className="formCheckboxes"
    >
      <div>
        <p>
          Payment in advance
        </p>
        <svg
          data-paymenttype="advance"
          data-testid="paymentInAdvance"
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1.1"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14 0h-12c-1.1 0-2 0.9-2 2v12c0 1.1 0.9 2 2 2h12c1.1 0 2-0.9 2-2v-12c0-1.1-0.9-2-2-2zM7 12.414l-3.707-3.707 1.414-1.414 2.293 2.293 4.793-4.793 1.414 1.414-6.207 6.207z"
          />
        </svg>
      </div>
      <div>
        <p>
          Payment on delivery
        </p>
        <svg
          data-paymenttype="delivery"
          data-testid="paymentOnDelivery"
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1.1"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14 0h-12c-1.1 0-2 0.9-2 2v12c0 1.1 0.9 2 2 2h12c1.1 0 2-0.9 2-2v-12c0-1.1-0.9-2-2-2zM7 12.414l-3.707-3.707 1.414-1.414 2.293 2.293 4.793-4.793 1.414 1.414-6.207 6.207z"
          />
        </svg>
      </div>
    </div>
    <h1>
      Total amount: 
      30.00
      $
    </h1>
    <a
      href="/order/choose_delivery/delivery_details/delivery_time"
      onClick={[Function]}
    >
      <button>
        Confirm order details
      </button>
    </a>
  </div>
</div>
`;
